src/Apps/Search/Routes/SearchResultsArtworks.tsx

```diff
@@ -19,11 +20,20 @@ interface SearchResultsRouteProps {
 export const SearchResultsArtworksRoute: React.FC<SearchResultsRouteProps> = props => {
   const { match } = useRouter()
   const { userPreferences } = useSystemContext()
+  const [searchFitlerKey, setSearchFilterKey] = useState(
+    match.location.query.term
+  )
   const { viewer } = props
   const { sidebar } = viewer
 
+  useEffect(() => {
+    // refresh artwork filter on query change
+    setSearchFilterKey(match.location.query.term)
+  }, [match.location.query.term])
+
   return (
     <ArtworkFilter
+      key={searchFitlerKey}
```

Comment:
**question:** It would not have occurred to me to use a `key` outside of a list context; would be helpful to understand the rationale for this. I take it that this forces the component to essentially be remounted when the url (and therefore the key) changes, but why is this necessary, especially given the presence of L39 below? 🤔

Reply:
When you reset the key, it will also also refreshes all internal state. No new key, state internal to components via `useState` is preserved across renders. But as soon as a parent key changes, each of those calls get recomputed. Note that this can be expensive. 

There's the question of whether `match.location.subFooVariables` as returned from `useRouter` is internally triggering its own state change when those values update. I would expect this to be the case, as you mention @anandaroop! but maybe that doesn't.